core/src/main/java/org/infinispan/interceptors/impl/EntryWrappingInterceptor.java

```diff
@@ -114,16 +115,22 @@
          FlagBitSets.SKIP_OWNERSHIP_CHECK | FlagBitSets.CACHE_MODE_LOCAL;
    private boolean totalOrder;
 
+   private void addVersionRead(InvocationContext rCtx, AbstractDataCommand dataCommand) {
+      // The entry must be in the context
+      CacheEntry cacheEntry = rCtx.lookupEntry(dataCommand.getKey());
+      cacheEntry.setSkipLookup(true);
+      if (isVersioned && ((MVCCEntry) cacheEntry).isRead()) {
+         addVersionRead((TxInvocationContext) rCtx, cacheEntry, dataCommand.getKey());
+      }
+   }
+
    private final InvocationSuccessAction dataReadReturnHandler = (rCtx, rCommand, rv) -> {
       AbstractDataCommand dataCommand = (AbstractDataCommand) rCommand;
 
       if (rCtx.isInTxScope() && useRepeatableRead) {
-         // The entry must be in the context
-         CacheEntry cacheEntry = rCtx.lookupEntry(dataCommand.getKey());
-         cacheEntry.setSkipLookup(true);
-         if (isVersioned && ((MVCCEntry) cacheEntry).isRead()) {
-            addVersionRead((TxInvocationContext) rCtx, cacheEntry, dataCommand.getKey());
-         }
+         // This invokes another method as this is only done with a specific configuration and we want to inline
+         // the notifier below
```

Comment:
It is pretty simple and from what I saw it was pretty consistently unable to inline this entire method, because it would make the caller too large. This would help inline when not using repeatable read, which seems good to me.